### PR TITLE
Front Continuous Integration 

### PR DESCRIPTION
**Pull request by StephaneGrabsia**
This PR tries to implement the github worflow for the front. It replaces the previous PR which was way too long and difficult to understand.

This PR defines a github action that lints all the .js files with eslinter, and then run the tests declared thanks to jest.

To test this PR manually you can:

Create a test for a React function
Commit it and push it upstream on this repo
Look if all the tests are OK

### CODE REVIEW

File: front/jest.config.js [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+  
+    // Indicates whether each individual test should be reported during the run
+    // verbose: undefined,
+  
+    // An array of regexp patterns that are matched against all source file paths before re-running tests in watch mode
+    // watchPathIgnorePatterns: [],
+  
+    // Whether to use watchman for file crawling
+    // watchman: true,
+  };
```
Comment (sachabinder) [inline front/jest.config.js:202]: why all this comments ?

File: front/package.json [hunk 2]
```diff
@@ -32,5 +39,14 @@
...
     "webpack-dev-server": "^4.11.1"
+  },
+  "dependencies": {
+    "eslint-config-prettier": "^8.5.0",
+    "eslint-import-resolver-jsconfig": "^1.1.0",
+    "eslint-plugin-import": "^2.26.0",
+    "eslint-plugin-prettier": "^4.2.1",
+    "jest": "^29.3.1",
+    "jest-environment-jsdom": "^29.3.1",
+    "prettier": "^2.8.0"
```
Comment (sachabinder) [inline front/package.json]: put that in dev dependancies and see if it works or not

**Review by sachabinder — DISMISSED** [review]
Good job 💯

File: front/jest.config.js (hunk 1 above)
Comment (StephaneGrabsia) [reply front/jest.config.js:202]: Created automatically by jest

**Review by sachabinder — APPROVED** [review]

**Pull request merged**